recipes/z3/all/conanfile.py

```diff
@@ -70,29 +83,56 @@ def _configure_cmake(self):
         self._cmake.configure(build_folder=self._build_subfolder)
         return self._cmake
 
+    @property
+    def _compilers_minimum_version(self):
+        return {
+            "gcc": "7",
+            "Visual Studio": "15.7",
+            "clang": "5",
+            "apple-clang": "10",
+        }
+
+    def validate(self):
+        if tools.Version(self.version) >= tools.Version("4.8.11"):
+            if self.settings.compiler.get_safe("cppstd"):
+                tools.check_min_cppstd(self, "17")
+            compiler = self.settings.compiler
+            min_version = self._compilers_minimum_version.get(str(compiler), False)
+            if min_version:
+                if tools.Version(compiler.version) < min_version:
+                    raise ConanInvalidConfiguration(
+                        f"{self.name} requires C++17, which {compiler} {compiler.version} does not support.")
+            else:
+                self.output.info(
+                    f"{self.name} requires C++17. Your compiler is unknown. Assuming it supports C++17.")
+
     def build(self):
         for patch in self.conan_data.get("patches", {}).get(self.version, []):
             tools.patch(**patch)
-        tools.save(os.path.join(self._build_subfolder, "gmp.h"), textwrap.dedent("""\
-            #pragma once
-            #include <mpir.h>
-            """))
+
+        if self.options.multiprecision == "mpir":
+            tools.save(os.path.join(self._build_subfolder, "gmp.h"), textwrap.dedent("""\
+                #pragma once
+                #include <mpir.h>
+                """))
+
         cmake = self._configure_cmake()
         cmake.build()
 
     def package(self):
         self.copy("LICENSE.txt", src=self._source_subfolder, dst="licenses")
         cmake = self._configure_cmake()
         cmake.install()
-
+        tools.rmdir(os.path.join(self.package_folder, "lib", "pkgconfig"))
         tools.rmdir(os.path.join(self.package_folder, "lib", "cmake"))
 
     def package_info(self):
         self.cpp_info.set_property("cmake_file_name", "Z3")
         self.cpp_info.set_property("cmake_target_name", "z3::libz3")
 
         # TODO: back to global scope in conan v2 once cmake_find_package_* generators removed
-        self.cpp_info.components["libz3"].libs = ["libz3" if self.settings.os == "Windows" else "z3"]
+        self.cpp_info.components["libz3"].libs = [
+            "libz3" if self.settings.os == "Windows" else "z3"]
         if not self.options.shared:
             if self.settings.os in ["Linux", "FreeBSD"]:
                 self.cpp_info.components["libz3"].system_libs.append("pthread")
```

Comment:
```suggestion
                self.cpp_info.components["libz3"].system_libs.append("pthread", "m")
```

Reply:
What does that change? Are you sure that works on Windows and Mac?

I can try anyways and see if the buildfarms are happy.

Reply:
Mac and Windows does not matter, it a change only for Linux and FreeBSD. What changed: it was introduced a CI job which detects linked system libraries and are not listed in the recipe. The libm missing.

Reply:
Oh yes my bad I read too fast. Thanks for the heads up.

Reply:
Had to write this way: `.system_libs.extend(["pthread", "m"])`